sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor{TPartition}.cs

```diff
@@ -29,7 +29,9 @@ namespace Azure.Messaging.EventHubs.Primitives
     ///
     /// <typeparam name="TPartition">The context of the partition for which an operation is being performed.</typeparam>
     ///
+#pragma warning disable CA1001 // Types that own disposable fields should be disposable
```

Comment:
I really wish there was a way to do these that didn't make the code look like swiss cheese.  That said, I'm not aware of any better approach, so I'll keep my sad to myself.

Reply:
I can move some of these into the use of `SuppressMessage` attributes, which might make things more palatable.